integration-tests/kubernetes-client/src/main/java/io/quarkus/it/kubernetes/client/Pods.java

```diff
@@ -28,8 +28,8 @@ public Pods(KubernetesClient kubernetesClient) {
     @GET
     @Produces(MediaType.APPLICATION_JSON)
     @Path("/{namespace}")
-    public List<Pod> pods(@PathParam("namespace") String namespace) {
-        return kubernetesClient.pods().inNamespace(namespace).list().getItems();
+    public Response pods(@PathParam("namespace") String namespace) {
+        return Response.ok(kubernetesClient.pods().inNamespace(namespace).list().getItems()).build();
```

Comment:
I think we should keep it as is as it's probably what users will do. At least, we shouldn't see that as solving the issue.

Reply:
I don't agree, at least not in the context of the integration test. The fact that we were returning Pods means that the entire hierarchy was being registered for reflection, which we definitely want to avoid

Reply:
Also if people are looking at integration tests as a model to create their applications, then all the more reason to show them the "better" way of doing this for this use case

Reply:
I don't know.

I mean: we need them to be registered for reflection somehow, don't we? Otherwise it won't work well in native image. If our REST-based registration logic doesn't work very well, maybe we need to reconsider it globally?

Reply:
The used Kubernetes Model classes are already registered by the Kubernetes extension (weakly) - and that is why the native test works :)

Reply:
Yes I know that. What I'm wondering is: should we generalize that or not? So that the model can be used in the REST API calls without thinking twice about it.

I'm unsure about the consequences thus the reason I ask.

Reply:
You can use it now without having to do anything. The use of `Pod` as a result type was really an oversight on my part when I created the integration test, it shouldn't have been done that way because it basically shadows the reflection registration that the Kubernetes extensions does by bringing the RESTEasy reflection generation in the mix

Reply:
Users will do that. You can be sure of that. So should we change the RESTEasy reflection registration to be weak too?

That's what I'm trying to ask since the beginning :).

Reply:
We can certainly do that, but I think it should be a following PR. We would need to do introduce a new Build Item that would be consumed by RESTEasy - or maybe it needs to be even more generalized

Reply:
Furthermore, we haven't had any users complain so far, so I maybe we shouldn't add extra complexity unless it's necessary